### PR TITLE
enable strict typescript

### PR DESCRIPTION
**Pull request by jaysonboyer**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,15 +1,22 @@
 {
   "scripts": {
-    "build": "ncc build ./src/index.ts --source-map"
+    "build": "ncc build ./src/index.ts --source-map",
+    "lint": "eslint --ext .js,.ts ."
```
Comment (tommymorgan) [inline package.json:4]: Is there any reason not to move the extensions into the .eslintrc?

File: src/increment-version.ts [hunk 2]
```diff
@@ -31,7 +31,9 @@ export function incrementVersion() {
       core.setOutput("new-version", newVersion);
       console.log({ cleanVersion, releaseType, newVersion });
     }
-  } catch (e) {
-    core.setFailed(e.message);
+  }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  catch (e: any) {
```
Comment (tommymorgan) [inline src/increment-version.ts]: Catching errors in TypeScript is a bit tedious, but you can do something like the following if you want all the typed goodness.

```typescript
try {
    // foo
} catch (e) {
    let msg = 'unknown error';

    if (typeof e === 'string') {
        msg = e;
    } else if (e instanceof Error) {
        msg = e.message;
    }

    core.setFailed(msg);
}
```

Comment (jaysonboyer) [reply src/increment-version.ts]: tedious but explicit and I like that.  code changed

File: package.json (hunk 1 above)
Comment (jaysonboyer) [reply package.json]: It doesn't look like it is part of the standard .eslintrc but it is part of the experimental eslint.config.js. I like the idea of pulling it into the config file but I will probably do that as a fast follow.
https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new#configuration-file

**Pull request merged**